Mathlib/CategoryTheory/Triangulated/Opposite.lean

```diff
@@ -0,0 +1,170 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Shift.Opposite
+import Mathlib.CategoryTheory.Shift.Pullback
+import Mathlib.CategoryTheory.Triangulated.Triangulated
+import Mathlib.Tactic.Linarith
+
+/-!
+# The (pre)triangulated structure on the opposite category
+
+In this file, we shall construct the (pre)triangulated structure
+on the opposite category `Cᵒᵖ` of a (pre)triangulated category `C`.
+
+The shift on `Cᵒᵖ` is obtained by combining the constructions in the files
+`CategoryTheory.Shift.Opposite` and `CategoryTheory.Shift.Pullback`.
+When the user opens `CategoryTheory.Pretriangulated.Opposite`, the
+category `Cᵒᵖ` is equipped with the shift by `ℤ` such that
+shifting by `n : ℤ` on `Cᵒᵖ` corresponds to the shift by
+`-n` on `C`. This is actually a definitional equality, but the user
+should not rely on this, and instead use the isomorphism
+`shiftFunctorOpIso C n m hnm : shiftFunctor Cᵒᵖ n ≅ (shiftFunctor C m).op`
+where `hnm : n + m = 0`.
+
+Some compatibilities between the shifts on `C` and `Cᵒᵖ` are also expressed through
+the equivalence of categories `opShiftFunctorEquivalence C n : Cᵒᵖ ≌ Cᵒᵖ` whose
+functor is `shiftFunctor Cᵒᵖ n` and whose inverse functor is `(shiftFunctor C n).op`.
+
+-/
+
+namespace CategoryTheory
+
+open Category Limits Preadditive ZeroObject
+
+variable (C : Type*) [Category C]
+
+namespace Pretriangulated
+
+variable [HasShift C ℤ]
+
+namespace Opposite
+
+/-- As it is unclear whether the opposite category `Cᵒᵖ` should always be equipped
+with the shift by `ℤ` such that shifting by `n` on `Cᵒᵖ` corresponds to shifting
+by `-n` on `C`, the user shall have to do `open CategoryTheory.Pretriangulated.Opposite`
+in order to get this shift and the (pre)triangulated structure on `Cᵒᵖ`. -/
+
+private abbrev OppositeShiftAux :=
+  PullbackShift (OppositeShift C ℤ)
+    (AddMonoidHom.mk' (fun (n : ℤ) => -n) (by intros; dsimp; linarith))
+
+/-- The category `Cᵒᵖ` is equipped with the shift such that the shift by `n` on `Cᵒᵖ`
+corresponds to the shift by `-n` on `C`. -/
+noncomputable scoped instance : HasShift Cᵒᵖ ℤ :=
+  (inferInstance : HasShift (OppositeShiftAux C) ℤ)
+
+instance [Preadditive C] [∀ (n : ℤ), (shiftFunctor C n).Additive] (n : ℤ) :
```

Comment:
Should this also be a scoped instance?

Reply:
I had tried, but the linter complained...

Reply:
In that case, let's kick this on the queue.